src/auth/auth.ts

```diff
@@ -0,0 +1,5 @@
+export default interface Auth {
+    // Hook into axios and intercept requests to add Auth tokens
+    // https://masteringjs.io/tutorials/axios/interceptors
+    initialize(): void;
+}
```

Comment:
This interface doesn't really communicate it's intent. It's just a method with no parameters and no return value.

Also, using interceptors on the "global" axios object is messy since it's hard to trace what interceptors are attached.

I think it'd be better to use an [axios instance](https://github.com/axios/axios#creating-an-instance) instead(also has other benefits like being able to specify a baseURL)

Then you can have an interface method like:
```js
attachInterceptor(axiosInstance: AxiosInstance): void
```

Reply:
updated